src/core/pointcloud/qgspointcloudlayerrenderer.cpp

```diff
@@ -715,6 +735,14 @@ void QgsPointCloudLayerRenderer::renderTriangulatedSurface( QgsPointCloudRenderC
   painter->drawImage( 0, 0, img );
 }
 
+Qgis::MapLayerRendererFlags QgsPointCloudLayerRenderer::flags() const
+{
+  // when rendering as triangles we still want to show temporary incremental renders as points until
+  // the final triangulated surface is ready, which may be slow
+  // So we request here a preview render image for the temporary incremental updates:
+  return Qgis::MapLayerRendererFlag::RenderPartialOutputs | Qgis::MapLayerRendererFlag::RenderPartialOutputOverPreviousCachedImage;
```

Comment:
Shouldn't this be dependent on whether render as triangles is active?

Reply:
True, since we can guarantee that `mRenderer` is only set in the constructor